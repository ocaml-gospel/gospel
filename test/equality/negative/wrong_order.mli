type 'a t

val f : 'a t -> 'a t -> ('a -> 'a -> bool) -> bool
(*@ equality *)
