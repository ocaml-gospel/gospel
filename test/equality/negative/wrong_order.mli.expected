
*******************************
********** Parsed file ********
*******************************
type 'a t
val f : 'a t -> 'a t -> ('a -> 'a -> bool) -> bool[@@gospel {| equality |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type 'a t
  

val f : 'a t -> 'a t -> ('a -> 'a -> bool) -> bool
(*@ equality
     *)
File "wrong_order.mli", lines 3-4, characters 0-25:
Error: Type checking error: type is incompatible with equality clause
